### PR TITLE
Add back appstoreenabled config switch

### LINKED ISSUE #2959: Disabled appstore in config.php still access the store
### Steps to reproduce
1. Add 'appstoreenabled' => false to config.php
2. Access https://cloudurl.com/settings/apps

### Expected behaviour

Only apps that exits on the filesystem should be installable. No other category than "Enabled" and "Not enabled" should be shown. (Like in Nextcloud 10)

### Actual behaviour

Many categories are shown like Files, Multimedia ... 
All Apps from the store are shown. 

### Server configuration
**Nextcloud version:** (see Nextcloud admin page)
Version 11.0.0 (git)

**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh

**Where did you install Nextcloud from:**
GIT

```

### PR DESCRIPTION
**Pull request by Gomez**
Will fix https://github.com/nextcloud/server/issues/2959

### CODE REVIEW

File: lib/private/App/AppStore/Fetcher/Fetcher.php [hunk 1]
```diff
@@ -69,31 +69,37 @@ public function __construct(IAppData $appData,
...
 	 */
 	protected function fetch($ETag, $content) {
-		$options = [];
 
-		if ($ETag !== '') {
-			$options['headers'] = [
-				'If-None-Match' => $ETag,
-			];
-		}
+		$systemConfig = \OC::$server->getSystemConfig();
```
Comment (ChristophWurst) [inline lib/private/App/AppStore/Fetcher/Fetcher.php]: should be injected via the constructor

File: lib/private/App/AppStore/Fetcher/Fetcher.php [hunk 2]
```diff
@@ -69,31 +69,37 @@ public function __construct(IAppData $appData,
...
+		$appstoreenabled = $systemConfig->getValue('appstoreenabled', true);
+		$responseJson = [];
+		if (!$appstoreenabled) {
```
Comment (ChristophWurst) [inline lib/private/App/AppStore/Fetcher/Fetcher.php]: simply return an empty array if  ``$appstoreenabled`` equals true, as that is the rare case. Also please rename the variable to camel case.

File: lib/private/App/AppStore/Fetcher/Fetcher.php [hunk 3]
```diff
@@ -102,43 +108,50 @@ protected function fetch($ETag, $content) {
...
-					return $jsonBlob['data'];
-				}
-
-				if (isset($jsonBlob['ETag'])) {
-					$ETag = $jsonBlob['ETag'];
-					$content = json_encode($jsonBlob['data']);
+	public function get() {
+		$systemConfig = \OC::$server->getSystemConfig();
+		$appstoreenabled = $systemConfig->getValue('appstoreenabled', true);
+		if (!$appstoreenabled) {
```
Comment (ChristophWurst) [inline lib/private/App/AppStore/Fetcher/Fetcher.php]: same here. Having the app store disabled is probably the exception, so do that in the if-block and return immediately.

File: lib/private/App/AppStore/Fetcher/Fetcher.php [hunk 4]
```diff
@@ -102,43 +108,50 @@ protected function fetch($ETag, $content) {
...
-				if((int)$jsonBlob['timestamp'] > ($this->timeFactory->getTime() - self::INVALIDATE_AFTER_SECONDS) &&
-					isset($jsonBlob['ncversion']) && $jsonBlob['ncversion'] === $this->config->getSystemValue('version', '0.0.0')) {
-					return $jsonBlob['data'];
-				}
-
-				if (isset($jsonBlob['ETag'])) {
-					$ETag = $jsonBlob['ETag'];
-					$content = json_encode($jsonBlob['data']);
+	public function get() {
+		$systemConfig = \OC::$server->getSystemConfig();
```
Comment (ChristophWurst) [inline lib/private/App/AppStore/Fetcher/Fetcher.php]: should also be injected

**Review by ChristophWurst — DISMISSED** [review]

**Comment by ChristophWurst** [comment]
whoops, did not see the "WIP" in the title. Sorry for reviewing this too early 🙈

**Comment by MorrisJobke** [comment]
Let me fix this.

**Review by MorrisJobke — APPROVED** [review]
I tested this and it works now 👍 I added some unit tests and used the injected config object.

**Comment by MorrisJobke** [comment]
@Gomez @ChristophWurst @nickvergessen @LukasReschke Please review - best use the whitespace cleaned diff: https://github.com/nextcloud/server/pull/3287/files?w=1

**Comment by MorrisJobke** [comment]
@Gomez Thanks for starting this and sorry for hijacking this PR, but I just wanted to have it fixed 😉

**Comment by nickvergessen** [comment]
I inverted the logic so the code is not indented once more.
Should also be small enough for a backport?

**Comment by Gomez** [comment]
Tested and works. Thanks @MorrisJobke! :+1:

**Review by nickvergessen — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@karlitschek We should backport this to stable11

**Comment by karlitschek** [comment]
yes. definitely backport.

**Comment by nickvergessen** [comment]
@Gomez can you create a backport against stable11 please?
Thanks!

**Pull request merged**